### PR TITLE
Parser add more parser methods

### LINKED ISSUE #33: Add More Parsers to Parser Class
Add more validity checks for user inputs: the index for income and expenditure.

### PR DESCRIPTION
**Pull request by 1szheng**
Fixes #33

### CODE REVIEW

File: src/main/java/seedu/planitarium/parser/Parser.java [hunk 1]
```diff
@@ -115,12 +129,53 @@ public static void isValidMoney(String amount) throws NumberFormatException {
...
+     * @throws NumberFormatException if amount is not a valid integer.
      * @throws IndexOutOfBoundsException if provided index is out of bounds.
      */
     public static void isValidUserIndex(String userIndex) throws NumberFormatException, IndexOutOfBoundsException {
         int checkIndex = Integer.parseInt(userIndex);
-        if (checkIndex < MIN_USER_INDEX || checkIndex >= PersonList.getNumberOfMembers()) {
+        if (checkIndex < MIN_USER_INDEX) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (checkIndex >= PersonList.getNumberOfMembers()) {
```
Comment (Teanweijun) [inline src/main/java/seedu/planitarium/parser/Parser.java:140]: I think it'll be more readable to have this as it's own function. Also allows for better SLAP

File: src/main/java/seedu/planitarium/parser/Parser.java [hunk 2]
```diff
@@ -115,12 +129,53 @@ public static void isValidMoney(String amount) throws NumberFormatException {
...
+     * @throws NumberFormatException if index is not a valid integer.
+     * @throws IndexOutOfBoundsException if provided index is out of bounds.
+     */
+    public static void isValidExpenditureIndex(String expenditureIndex, Person person)
+            throws NumberFormatException, IndexOutOfBoundsException {
+        int checkIndex = Integer.parseInt(expenditureIndex);
+        if (checkIndex < MIN_EXPENDITURE_INDEX) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (checkIndex >= person.getNumberOfExpenditures()) {
```
Comment (Teanweijun) [inline src/main/java/seedu/planitarium/parser/Parser.java:159]: Same as above

File: src/main/java/seedu/planitarium/parser/Parser.java [hunk 3]
```diff
@@ -115,12 +129,53 @@ public static void isValidMoney(String amount) throws NumberFormatException {
...
+     *
+     * @param incomeIndex Person's income lookup index.
+     * @param person      Person who has income.
+     * @throws NumberFormatException if index is not a valid integer.
+     * @throws IndexOutOfBoundsException if provided index is out of bounds.
+     */
+    public static void isValidIncomeIndex(String incomeIndex, Person person)
+            throws NumberFormatException, IndexOutOfBoundsException {
+        int checkIndex = Integer.parseInt(incomeIndex);
+        if (checkIndex < MIN_INCOME_INDEX) {
```
Comment (Teanweijun) [inline src/main/java/seedu/planitarium/parser/Parser.java:175]: One more instance here

**Review by Teanweijun — COMMENTED** [review]
Looks good! Just need to abstract out some functions and it'll be good to merge

**Review by hlwang56 — APPROVED** [review]
Looking good

**Pull request merged**